A variety of bugfixes and usability improvements related to Struggle

magroader opened the pull request:
- It no longer shows -/+ buttons (which are buggy anyway, see #407
- it no longer says Unlimited/99 on the move details screen
- It no longer is available to choose when choosing a move for a Pokemon (though if the Pokemon already knows it, it will be available to swap out)
- It shows up on every Pokemon's move list on the party screen (at the bottom)

Also, no move (including Struggle) will show the "Reset" button on the move details if it's already at max PP.

#408

magroader commented on pokedex/pokemon.lua:
```diff
@@ -569,10 +595,6 @@ function M.get_move_pp_max(pokemon, move)
 	return movedex.get_move_pp(move) + pp_extra
 end
 
-function M.get_move_index(pokemon, move)
```
This was unused

Jerakin commented:
Trying to bring the info about the Struggle move that was added automatically soft-crashes the app
```
ERROR:SCRIPT: /screens/popups/move_info/move_info.gui_script:56: attempt to index field 'data' (a nil value)
stack traceback:
	/screens/popups/move_info/move_info.gui_script:56: in function </screens/popups/move_info/move_info.gui_script:22>
	[C]: in function 'resume'
	/monarch/monarch.lua:893: in function 'on_message'
	/monarch/screen_factory.script:43: in function </monarch/screen_factory.script:35>
ERROR:GAMESYS: Error when initializing gui component: -2.
```

magroader commented:
Not sure how to reply to your comment, but I fixed the issue with details pane of Struggle.

Also fixed an issue where using Reset from that details pane (for other moves) was not refreshing the PP in the main section. This was happening in master!
